instance/cloudServerCreate.go

```diff
@@ -196,6 +195,26 @@ func (ci *Client) CloudServerCreate(params *CloudServerCreateParams) (string, er
 		createArgs["image_id"] = params.ImageId
 	}
 
+	// when creating with a config-id, adjust the Type param for bare-metal types if blatantly wrong
+	var configDetails apiTypes.CloudConfigDetails
+	if params.ConfigId != -1 {
+		if err := ci.CallLwApiInto("bleed/storm/config/details",
+			map[string]interface{}{"id": params.ConfigId}, &configDetails); err != nil {
+			return "", err
+		}
+		if configDetails.Category == "bare-metal" {
+			if strings.Contains(params.Type, "SS.VPS") {
```

Comment:
What happens if the customer gives a type of `SS.VM` but they don't have the windows stuff correct?

Reply:
as written when that happens it would return an error about `$WINDOWS_TEMPLATE_NAME` not being permitted on `SS.VM`. So that's a fair point. I've adjusted it in this commit 6bf8f56a2fa4d98f99e3ca6e2560123a0b782d4a to always set either `SS.VM` or `SS.VM.WIN` in such a case